planet/src/main/webapp/places.js

```diff
@@ -123,51 +135,76 @@ function addPlaceDetails(places) {
         } else {
             bounds.extend(place.geometry.location);
         }
-    });   
+
+        // Get place id
+        let placeId = place['place_id']; 
+        let url = 'https://maps.googleapis.com/maps/api/place/details/json?placeid=' + placeId + '&key=AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns&fields=photo';
+        
+        // Fetch url and add photo reference to a list
+        promises = [];
+        promises.push(fetch(url).then(response => response.json()).then(function(response) {
+            photoReference = response.result.photos[0].photo_reference;
+            imgList.push(photoReference);
```

Comment:
I don't see anything that guarantees that the ordering of the entries in imgList matches the ordering in placeInfo.

Consider using a map here to look up the image for the place.